apps/api/src/persistence/database/arango-query-runner.ts

```diff
@@ -0,0 +1,35 @@
+import { HasId } from '@coscrad/api-interfaces';
+import { isDeepStrictEqual } from 'util';
+import { ICoscradQueryRunner } from '../migrations/coscrad-query-runner.interface';
+import { ArangoDatabase } from './arango-database';
+import { ArangoCollectionId } from './collection-references/ArangoCollectionId';
+import { ArangoDatabaseProvider } from './database.provider';
+import { DatabaseDocument } from './utilities/mapEntityDTOToDatabaseDTO';
+
+export class ArangoQueryRunner implements ICoscradQueryRunner {
+    private readonly arangoDatabase: ArangoDatabase;
+
+    constructor(arangoDatabaseProvider: ArangoDatabaseProvider) {
+        this.arangoDatabase = arangoDatabaseProvider.getDBInstance();
+    }
+
+    async update<TOldDocument extends DatabaseDocument<HasId>, UNewDocument>(
+        collectionName: ArangoCollectionId,
+        calculateUpdate: (oldDocument: TOldDocument) => UNewDocument
+    ): Promise<void> {
+        const existingDocs = await this.arangoDatabase.fetchMany<TOldDocument>(collectionName);
+
+        const updates = existingDocs
+            .map((oldDocument) => ({
+                ...calculateUpdate(oldDocument),
+                // this ensures that the update can be matched to an existing doc
+                _key: oldDocument._key,
```

Comment:
Still can't figure out your TDD slack post, but I see you've moved to having the `oldDocument` here...

Reply:
Oh right. I didn't flow through the arg `oldDocument` to call `calculateUpdate` with. Hence it was trying to stick the function `calculateUpdate` itself into the updated document. Brain fart on my part. Fixed now.